### PR TITLE
[GraphView]refactor code (add_spouses / Animation class) & cleanup[gramps50]

### PR DESCRIPTION
**Pull request by vantu5z**
- add generation spinners to panel for quick switch.
- add_spouses moved to separate method.
- refactor code to make it more good-looking (width < 80 characters where it possible).
- fix comments to look uniform.
- code clean up (fix tabulation, remove unused methods, commented and fixed issues code).
- set descendant generation (1 .. 50) default 10.
- set ancestor generation (0 .. 50) default 3.
- added two spinner controls to the mini toolbar to show more/less generations for ancestors & descendants

### CODE REVIEW

**Comment by PQYPLZXHGF** [comment]
> some enhancement

What type of enhancement, does it change the way Graph View displays?

File: GraphView/graphview.py [hunk 1]
```diff
@@ -1552,145 +1540,129 @@ def set_current_list_desc(self, active_person):
...
+
+        # add all his(her) spouses recursively
+        if spouse_handle:
+            sp_person = self.database.get_person_from_handle(spouse_handle)
+
+        if sp_person:
+            for family_handle in sp_person.get_family_handle_list():
+                sp_family = self.database.get_family_from_handle(family_handle)
+
+                m_handle = sp_family.get_mother_handle()
```
Comment (vantu5z) [inline GraphView/graphview.py]: read mother_handle and father_handle just once (instead of 3 times).
may little increase speed.

**Comment by vantu5z** [comment]
@PQYPLZXHGF , no it should not change any visual aspect View display.
- `add_spouses` moved to separate method (need little test with multiple spouses)
- refactor code to make it more good-looking (width < 80 characters where it possible).
- fix comments to look uniform.
- code clean up (fix tabulation, remove unused methods, commented and fixed issues code).

File: GraphView/graphview.py [hunk 2]
```diff
@@ -684,61 +678,63 @@ def goto_active(self, button):
...
+
         # save transform scale
         self.transform_scale = parser.transform_scale
         self.set_zoom(self.scale)
 
         self.animation.update_items()
 
-        # Scroll to activ person without animation
-        self.goto_active(None)
+        # scroll to activ person without animation
```
Comment (PQYPLZXHGF) [inline GraphView/graphview.py]: > scroll to activ person without animation

activ > active

Comment (vantu5z) [reply GraphView/graphview.py]: thanks, I'll fix this

**Comment by sam-m888** [comment]
@vantu5z Seems quicker on my 10K family tree.   Only came across the following error that I've not been able to recreate so far, so here it is:

> $ ./Gramps.py 
> Warning: History: Exception occurred in callback function.
> Traceback (most recent call last):
>   File "/home/grampsdev/aaSandbox/gramps/gramps/gen/utils/callback.py", line 405, in emit
>     fn(*args)
>   File "/home/grampsdev/aaSandbox/gramps/gramps/gui/views/navigationview.py", line 199, in goto_active
>     self.goto_handle(active_handle)
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 261, in goto_handle
>     self.graph_widget.populate(self.get_active())
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 707, in populate
>     dot.build_graph(active_person)
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 1500, in build_graph
>     self.person_handles.update(self.find_descendants(active_person))
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 1562, in find_descendants
>     person_handles)
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 1587, in add_descendant
>     person_handles)
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 1589, in add_descendant
>     self.add_spouses(person, family, person_handles)
>   File "/home/grampsdev/.gramps/gramps50/plugins/GraphView/graphview.py", line 1609, in add_spouses
>     if sp_person:
> UnboundLocalError: local variable 'sp_person' referenced before assignment
>

**Comment by sam-m888** [comment]
@vantu5z One suggestion is now that the addon can show ancestors have it default to 3 ancestral generations back so people can at least be aware the addon can now do it. eg change to:
`('interface.graphview-ancestor-generations', 3),`
Alternatively add a two controls to the mini toolbar to show more ancestor or descendant generations ?

**Comment by vantu5z** [comment]
@sam-m888 Add spinners to panel. Do you mean something like this.

**Comment by sam-m888** [comment]
Thank you

**Comment by fxtmtrsine** [comment]
@vantu5z This is great 🥇  Thanks

@sam-m888 FYI I think this covers [6186](https://gramps-project.org/bugs/view.php?id=6186): Limit number of generations in Graph View

**Comment by sam-m888** [comment]
@fxtmtrsine Thank you, feature request 6186 has been updated and marked as
resolved.

**Comment by fxtmtrsine** [comment]
@vantu5z The options in Configure/Layout for ancestor & descendant generations are probably redundant and can be removed?

**Pull request merged**